snowpack/src/sources/local.ts

```diff
@@ -494,13 +498,10 @@ export class PackageSourceLocal implements PackageSource {
       const lineBullet = colors.dim(depth === 0 ? '+' : '└──'.padStart(depth * 2 + 1, ' '));
       let packageFormatted = spec + colors.dim('@' + packageVersion);
       const existingImportMapLoc = path.join(installDest, 'import-map.json');
-      const importMapHandle = await fs.open(existingImportMapLoc, 'r+').catch(() => null);
-
-      let existingImportMap: ImportMap | null = null;
-      if (importMapHandle) {
-        const importMapData = await importMapHandle.readFile('utf-8');
-        existingImportMap = importMapData ? JSON.parse(importMapData) : null;
-        await importMapHandle.close();
+      let existingImportMap: ImportMap | undefined = memoizedImportMap[packageName];
+      if (!existingImportMap && existsSync(existingImportMapLoc)) {
+        existingImportMap = JSON.parse(await fs.readFile(existingImportMapLoc, 'utf8'));
```

Comment:
Using `existsSync` and then `readFile` still scares me as the previous bug was that the file existed but was not open for writing (because currently being written) which caused it to throw at this line. Even if logic prevents the race we should still use `fs.open` here as that guarantees both that the file exists and that it's finished being written to.

Reply:
Oh you have a good point. Will change back, but probably keep the in-memory memoization for already-read files